apps/src/lab2/lab2Redux.ts

```diff
@@ -475,7 +485,17 @@ async function setUpAndLoadProject(
     }
   });
   projectManager.addSaveFailListener(() => dispatch(setProjectUpdatedError()));
-  return await projectManager.load();
+  // Figure out if we should reset to start sources. This happens if the url parameter
+  // ?reset=true is present and we have not already reset in this session.
+  // This parameter is only used by levelbuilders.
+  const url = new URL(document.URL);
+  const params = new URLSearchParams(url.search);
+  let resetToStartSources = false;
+  if (params.get('reset') && !alreadyResetHistory) {
```

Comment:
Did you consider managing this by modifying the URL directly after you've read this once? Something like

```
if (params.get('reset') === 'true') {
  // Remove the reset parameter from the URL
  params.delete('reset');
  // Update the URL in the browser without reloading the page
  window.history.replaceState({}, '', `${url.pathname}?${params.toString()}`);
  ...
}
```

It seems maybe a little simpler if the reset logic is straightforward and only needs to happen once without further checks?

Reply:
When I did that it forced the page to reload, which doesn't work because we won't have time to save the updated sources, so I think we'd end up not resetting the sources.

Reply:
👍 In that case, this approach LGTM, though I wouldn't hate getting a second set of eyes from @thomasoniii or someone else a little more familiar with the state management here!

Reply:
I think `window.history.pushState` can modify the URL without reloading the page.

Reply:
`pushState` was the trick! Much simpler now!